fairmq/shmem/Message.h

```diff
@@ -291,21 +295,28 @@ class Message final : public fair::mq::Message
             fManager.GetSegment(fMeta.fSegmentId);
             ShmHeader::IncrementRefCount(fManager.GetAddressFromHandle(fMeta.fHandle, fMeta.fSegmentId));
         } else { // unmanaged region
-            if (otherMsg.fMeta.fShared < 0) { // if UR msg is not yet shared
-                // TODO: minimize the size to 0 and don't create extra space for user buffer alignment
-                char* ptr = fManager.Allocate(2, 0);
-                // point the fShared in the unmanaged region message to the refCount holder
-                otherMsg.fMeta.fShared = fManager.GetHandleFromAddress(ptr, fMeta.fSegmentId);
-                // the message needs to be able to locate in which segment the refCount is stored
-                otherMsg.fMeta.fSegmentId = fMeta.fSegmentId;
-                // point this message to the same content as the unmanaged region message
-                fMeta = otherMsg.fMeta;
-                // increment the refCount
-                ShmHeader::IncrementRefCount(ptr);
+            if (otherMsg.fMeta.fShared < 0) {
+                fRegionPtr = fManager.GetRegionFromCache(otherMsg.fMeta.fRegionId);
+                if (fRegionPtr) {
+                    // create a reference couting object in for this message
+                    RefCount* rcPtr = fRegionPtr->MakeRefCount();
+                    // point the fShared in the unmanaged region message to the refCount object
+                    otherMsg.fMeta.fShared = fRegionPtr->HandleFromAddress(rcPtr);
+                    // point this message to the same content as the unmanaged region message
+                    fMeta = otherMsg.fMeta;
+                    // increment the refCount
+                    fRegionPtr->IncrementRefCount(rcPtr);
```

Comment:
```suggestion
                    rcPtr->Increment();
```

or am I overlooking any reason why this should be indirected through the `UnmanagedRegion` still?